From acda54cbc1985585a87a9e0a58a6d1da0623a40f Mon Sep 17 00:00:00 2001
From: Utkarsh Ayachit <utkarsh.ayachit@kitware.com>
Date: Sun, 19 Jul 2015 21:23:58 -0400
Subject: [PATCH 1/3] Fix interaction in vtkPVOrthographicSliceView.

The center of rotation wasn't being passed to the interactor style
correctly. Fixed that.

Also fixed the resizing of the axes grid. vtkPVOrthographicSliceView was
not calling superclass method which did the resizing of the axes grid.
---
 .../Rendering/vtkPVOrthographicSliceView.cxx              | 15 +++++++++++++++
 .../Rendering/vtkPVOrthographicSliceView.h                |  5 +++++
 ParaViewCore/ClientServerCore/Rendering/vtkPVRenderView.h |  4 ++--
 3 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.cxx b/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.cxx
index 6662834..b12fb10 100644
--- a/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.cxx
+++ b/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.cxx
@@ -380,6 +380,7 @@ void vtkPVOrthographicSliceView::SetSlicePosition(double x, double y, double z)
 //----------------------------------------------------------------------------
 void vtkPVOrthographicSliceView::UpdateCenterAxes()
 {
+  this->Superclass::UpdateCenterAxes();
   vtkBoundingBox bbox(this->GeometryBounds);
   double widths[3];
   bbox.GetLengths(widths);
@@ -564,6 +565,20 @@ void vtkPVOrthographicSliceView::SetTexturedBackground(int val)
 }
 
 //----------------------------------------------------------------------------
+void vtkPVOrthographicSliceView::SetCenterOfRotation(double x, double y, double z)
+{
+  this->Superclass::SetCenterOfRotation(x, y, z);
+  this->OrthographicInteractorStyle->SetCenterOfRotation(x, y, z);
+}
+
+//----------------------------------------------------------------------------
+void vtkPVOrthographicSliceView::SetRotationFactor(double factor)
+{
+  this->Superclass::SetRotationFactor(factor);
+  this->OrthographicInteractorStyle->SetRotationFactor(factor);
+}
+
+//----------------------------------------------------------------------------
 void vtkPVOrthographicSliceView::PrintSelf(ostream& os, vtkIndent indent)
 {
   this->Superclass::PrintSelf(os, indent);
diff --git a/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.h b/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.h
index 52f081d..f637b4b 100644
--- a/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.h
+++ b/ParaViewCore/ClientServerCore/Rendering/vtkPVOrthographicSliceView.h
@@ -104,6 +104,11 @@ protected:
   virtual void AboutToRenderOnLocalProcess(bool interactive);
   virtual void UpdateCenterAxes();
 
+  //*****************************************************************
+  // Forward to vtkPVOrthographicSliceView instances.
+  virtual void SetCenterOfRotation(double x, double y, double z);
+  virtual void SetRotationFactor(double factor);
+
   enum
     {
     SIDE_VIEW = 0,
diff --git a/ParaViewCore/ClientServerCore/Rendering/vtkPVRenderView.h b/ParaViewCore/ClientServerCore/Rendering/vtkPVRenderView.h
index db8caa0..11dd360 100644
--- a/ParaViewCore/ClientServerCore/Rendering/vtkPVRenderView.h
+++ b/ParaViewCore/ClientServerCore/Rendering/vtkPVRenderView.h
@@ -454,8 +454,8 @@ public:
 
   //*****************************************************************
   // Forward to vtkPVInteractorStyle instances.
-  void SetCenterOfRotation(double x, double y, double z);
-  void SetRotationFactor(double factor);
+  virtual void SetCenterOfRotation(double x, double y, double z);
+  virtual void SetRotationFactor(double factor);
 
   //*****************************************************************
   // Forward to vtkLightKit.
-- 
1.8.0.msysgit.0

